streamer/src/packet.rs

```diff
@@ -23,13 +31,20 @@ pub fn recv_from(obj: &mut Packets, socket: &UdpSocket, max_wait_ms: u64) -> Res
     socket.set_nonblocking(false)?;
     trace!("receiving on {}", socket.local_addr().unwrap());
     let start = Instant::now();
+
     loop {
         obj.packets.resize(
             std::cmp::min(i + NUM_RCVMMSGS, PACKETS_PER_BATCH),
             Packet::default(),
         );
-        match recv_mmsg(socket, &mut obj.packets[i..]) {
-            Err(_) if i > 0 => {
+        match recv_mmsg(socket, &mut obj.packets[i..], max_wait_ms) {
+            Err(e) if i > 0 => {
+                if e.kind() == io::ErrorKind::WouldBlock
+                    && (start.elapsed().as_micros() + RECV_BUFFER_DELAY_US as u128)
+                        < (max_wait_ms * 1_000) as u128
+                {
+                    sleep(Duration::from_micros(RECV_BUFFER_DELAY_US));
```

Comment:
In other testing I've done, I've noticed that it is hard (impossible?) to sleep below what the frequency of the system time-slice is. I believe that's around 100hz by default on linux which is a 1ms sleep.

I'm not sure that means we change it here, just trying to be aware of what the actual behavior might be.

Of course, this is configurable as well, so some kernels may have it higher frequency.